lib/CdeContext.tsx

```diff
@@ -1,83 +1,159 @@
+                                       collections,
+                                       config,
+                                       name,
+                                       callback,
+                                       children
+                                   }: PropsWithChildren<InitParams>) => {
+
+    let datasetSample;
+    // Process and validate datasetSample
+    try {
+        validateDatasetSample(rawDatasetSample);
+        datasetSample = mapDatasetSample(rawDatasetSample);
+    } catch (error) {
+        const message = error instanceof Error ? error.message : 'An unknown error occurred';
+        // TODO: Check if it's fine to throw or should we show an error in the UI
+        throw new Error(`Invalid dataset sample: ${message}`);
+    }
+
+    // Process and validate datasetMapping
+    let initialDatasetMapping;
+    let initialDatasetMappingHeader;
+    try {
+        validateDatasetMapping(rawDatasetMapping);
+        const datasetMappingData = mapStringTableToDatasetMapping(rawDatasetMapping);
+        initialDatasetMapping = datasetMappingData[0]
+        initialDatasetMappingHeader = datasetMappingData[1]
+    } catch (error) {
+        const message = error instanceof Error ? error.message : 'An unknown error occurred';
+        throw new Error(`Invalid dataset mapping: ${message}`);
+    }
+
+    const additionalDatasetMappings: DatasetMapping[] = [];
+    rawAdditionalDatasetMappings.forEach((additionalMapping, index) => {
+        try {
+            validateDatasetMapping(additionalMapping);
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            const [mappedAdditionalMapping, _] = mapStringTableToDatasetMapping(additionalMapping);
+            additionalDatasetMappings.push(mappedAdditionalMapping);
+        } catch (error) {
+            if (error instanceof Error) {
+                console.warn(`Skipping invalid additionalDatasetMapping at index ${index}: ${error.message}`);
+            } else {
+                console.warn(`Skipping invalid additionalDatasetMapping at index ${index}: Unknown error`);
+            }
+        }
+    });
 
+    const [datasetMapping, setDatasetMapping] = useState<DatasetMapping>(initialDatasetMapping);
+    const [datasetMappingHeader, setDatasetMappingHeader] = useState<string[]>(initialDatasetMappingHeader);
 
+    const [step, setStep] = useState(STEPS.HOME);
+    const [loadingMessage, setLoadingMessage] = useState<string | null>(null);
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
     const handleClose = () => {
-        callback(null);
-        setIsOpen(false)
+        setErrorMessage(null);
+        setLoadingMessage(null);
+        callback(datasetMapping)
     };
 
+    const handleUpdateDatasetMappingRow = (key: string, newData: CDE | CustomDictionaryField) => {
+        updateDatasetMappingRow(
```

Comment:
I did not test this function. I would prefer to do it when I the mapping step UI is complete